packages/flutter/test/material/bottom_sheet_test.dart

```diff
@@ -696,6 +696,40 @@ void main() {
     // the BottomNavigationBar.
     expect(tester.getBottomLeft(find.byType(BottomSheet)).dy, 600.0);
   });
+
+  testWidgets('Verify that route settings is passed during navigation',
+    (WidgetTester tester) async {
+    final GlobalKey<ScaffoldState> scaffoldKey = GlobalKey<ScaffoldState>();
+    const RouteSettings routeSettings =
+        RouteSettings(name: 'route_name', arguments: 'route_argument');
+
+    await tester.pumpWidget(MaterialApp(
+      home: Scaffold(
+        key: scaffoldKey,
+        body: const Center(child: Text('body')),
+      ),
+    ));
+
+    RouteSettings retreivedRouteSettings;
+
+    showModalBottomSheet<void>(
+      context: scaffoldKey.currentContext,
+      routeSettings: routeSettings,
+      builder: (BuildContext context) {
+        retreivedRouteSettings = ModalRoute.of(context).settings;
+        return Container(
+          child: const Text('BottomSheet'),
+        );
+      },
+    );
+
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
+
+    expect(retreivedRouteSettings.arguments, 'route_argument');
```

Comment:
since you already defined the routeSettings, can you directly check for identity?
`expect(retreivedRouteSettings, routeSettings)`